### DISCUSSION TITLE
Default Browser Banner Description string - Where does it really come from?

### DISCUSSION THREAD

**Discussion by gutley** [mozilla-mobile/firefox-ios#11785] [Q&A] [answered]
I'm updating my fork of the code to bring it in line with 104.1. I am trying to work out where the actual string for "Set links from websites, emails, and Messages to open automatically in Firefox." comes from. It seems it should be from HomeTabBanner.EvergreenMessage.HomeBannerDescription, but I don't think it is.

In my fork, I've modified `MZLocalizedString` to swap out any occurrences of "Firefox" with my own browser name E.g:

```swift
private func MZLocalizedString(_ key: String,
                               tableName: String? = nil,
                               value: String = "",
                               comment: String,
                               lastUpdated: StringLastUpdatedAppVersion
) -> String {
    var s = NSLocalizedString(key,
                             tableName: tableName,
                             bundle: Strings.bundle,
                             value: value,
                             comment: comment)
    s = s.replacingOccurrences(of: "Firefox", with: "My Browser")
    return s
}
```

This works great for every string with Firefox in it, except for the Default Browser banner description.

I even modified MZLocalizedString to just return "Say What?" and this is what I get:

![image](https://user-images.githubusercontent.com/2040690/188127867-4ea2f7d6-12a9-4f20-a8bc-8104ec97daea.png)

So it seems that those two strings are bypassing MZLocalizedString.

What's the story?

**Comment by lmarceau** [comment, marked as answer]
The home tab banner is configured with `Messages` coming from Nimbus. When a user is part of that experiment for banner, they will see different messages in there. They cannot be locally modified. I guess you could change where the MessageCardDataAdaptor takes it's input from, or remove the banner entirely.

**Reply by gutley** [reply]
Which poses the question though why the Default Browser banner is even related to the experiments code? Surely it's a fully fledged feature now.

Anyway - I have solved it locally by swizzling Bundle.localizedString to do my replacements in there.
